src/components/Books/Books.js

```diff
@@ -0,0 +1,66 @@
+import React, { useState, useCallback, useEffect } from 'react'; // eslint-disable-line no-unused-vars, prop-types
+import { useDebounce } from 'react-use';
+import { useMappedState, useDispatch } from 'redux-react-hook';
+import TextField from '@material-ui/core/TextField';
+import { withStyles } from '@material-ui/core';
+import InputAdornment from '@material-ui/core/InputAdornment';
+import SearchIcon from '@material-ui/icons/Search';
+import CircularProgress from '@material-ui/core/CircularProgress';
+import Item from '../Item/Item';
+import { fetchBooks } from '../../redux/actions/bookActions';
+
+const styles = {
+  searchInput: {
+    color: '#fff',
+  },
+};
+
+export const Books = props => {
+  const { classes } = props;
+  const [inputSearch, setInputSearch] = useState('');
+  const [performingSearch, setPerformingSearch] = useState(false);
+
+  const mapState = state => state.books;
+  const { books } = useMappedState(mapState);
+
+  const dispatch = useDispatch();
+  const fetchBooksCallback = useCallback(fetchBooks(inputSearch, dispatch));
+  useDebounce(fetchBooksCallback, 500, [inputSearch]);
+
+  useEffect(() => {
+    setPerformingSearch(false);
+  }, [books, inputSearch]);
```

Comment:
I don't think inputSearch is necessary as input. We should set the performing search to false whenever we get books, it shouldn't depend on the inputSearch changing right? also, if inputSearch changes, it also sets performing search to true on the next useEffect...

Reply:
Right now it's setting `performingSearch` to false and true on the same render. `useDebounce` is calling `fetchBooks` with an empty title on mounting, and the response, an empty array is updating books, causing this useEffect to trigger again and set `performingSearch` to false.

This causes a LoadingCircle to appear at mounting the component. My suggestion would be:

First: swap the useEffects:

```suggestion
  useEffect(() => {
    setPerformingSearch(true);
  }, [inputSearch]);

  useEffect(() => {
    setPerformingSearch(false);
  }, [books]);
```

Then I would add an if on `fetchBooks` action to not perform a call to orient-express if the title is empty and just set the books to `[]`. This way, if we delete completely the search bar, results will be cleared.

```suggestion
export const fetchBooks = (inputSearch, dispatch) => () => {
  if (inputSearch === '') {
    dispatch({
      type: FETCH_BOOKS,
      payload: [],
    });
  } else {
    axios
      .get(
        `${orientExpressBaseUrl}/search-book?title=%22${inputSearch}%22&from=1&to=3`,
      )
      .then(res => {
        dispatch({
          type: FETCH_BOOKS,
          payload: res.data,
        });
      });
  }
};
```